Lib/ensurepip/__init__.py

```diff
@@ -27,52 +17,36 @@
 _WHEEL_PKG_DIR = sysconfig.get_config_var('WHEEL_PKG_DIR')
 
 
-def _find_packages(path):
-    packages = {}
+def _find_wheel_pkg_dir_pip():
+    if _WHEEL_PKG_DIR is None:
+        return None
     try:
-        filenames = os.listdir(path)
+        filenames = os.listdir(_WHEEL_PKG_DIR)
     except OSError:
         # Ignore: path doesn't exist or permission error
-        filenames = ()
+        return None
```

Comment:
@AA-Turner so the reason I was using a `LookupError` was that it's better structurally and using sentinels like `None` introduces a mental burden on the callers forcing them not to forget to check for a magic value of a different type with a special meaning. This reminds me of ancient languages or golang where this same burden looks like having to check some integer return value for whether it's `-1` to decide if it's usable.

Reply:
No strong opinions here; I see both patterns. The caller must do something with either pattern -- either check for ``None`` or catch the error. No objections to go back to the error pattern if you'd prefer.

Reply:
I like forcing the caller to react rather than hoping that side won't forget... Anyway, I included this change as a part of the pathlib refactoring @ https://github.com/python/cpython/commit/b2a69a121a82d5abc8b99eed2492c4602d9fb91b. Marking the thread as resolved.